### PR TITLE
refactor: update HSBK RGB conversion to float (0.0-1.0)

### PR DESCRIPTION
**Pull request by Djelibeybi**
## Summary

- Update `HSBK.from_rgb()` to accept float values (0.0-1.0) instead of int (0-255)
- Update `HSBK.to_rgb()` to return `tuple[float, float, float]` (0.0-1.0) instead of `tuple[int, int, int]`
- Change hue type from `int` to `float` for exact round-trip fidelity (properties still return rounded values for display/equality)
- Convert all ~148 `Colors` enum entries from int to float RGB values
- Replace fireworks private `_hsb_to_rgb()`/`_rgb_to_hsb()` helpers with `HSBK` methods (removes ~70 lines)
- Update all callers across source, tests, examples, and documentation

## Motivation

The int-based (0-255) RGB conversion quantized values, causing 84% of colors to produce a mismatch on round-trip with errors up to 3/255. Since `colorsys` already works with 0.0-1.0 floats internally, the normalization and denormalization steps were pure overhead that introduced precision loss.

## Test plan

- [x] All 2425 tests passing
- [x] `uv run pyright` — 0 errors
- [x] `uv run ruff check . && uv run ruff format --check .` — clean
- [x] RGB round-trip error is within floating-point epsilon (~10⁻¹⁵)
- [x] Fireworks effect behavior preserved (uses HSBK methods instead of private helpers)

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: conductor/tracks/hsbk-rgb-float_20260318/plan.md [hunk 1]
```diff
@@ -15,68 +15,68 @@ Update `from_rgb()` and `to_rgb()` signatures and their direct tests.
...
+- [x] Task 1.2: Update `HSBK.from_rgb()` to accept `float` (0.0–1.0), remove int validation and `/ 255` normalization, remove rounding of hue/sat/bri
+- [x] Task 1.3: Update `HSBK.to_rgb()` to return `tuple[float, float, float]` (0.0–1.0), remove `* 255` denormalization and `int(round())`
+- [x] Task 1.4: Update docstrings and examples on both methods
 
 ### Verification
 
-- [ ] `test_color.py` passes with float-based assertions
-- [ ] Round-trip error is within floating-point epsilon
+- [x]`test_color.py` passes with float-based assertions
+- [x]Round-trip error is within floating-point epsilon
```
[the remark this answers is not in the document]
Comment (Djelibeybi) [reply conductor/tracks/hsbk-rgb-float_20260318/plan.md]: Fixed in `dd6454f` — added missing space after `[x]` on lines 25-26.

File: conductor/tracks/hsbk-rgb-float_20260318/plan.md [hunk 2]
```diff
@@ -15,68 +15,68 @@ Update `from_rgb()` and `to_rgb()` signatures and their direct tests.
...
-- [ ] Task 2.2: Verify `Colors` enum values are correct (spot-check primaries, grays, edge cases)
+- [x] Task 2.1: Convert all `Colors` enum entries from `from_rgb(int, int, int)` to `from_rgb(float, float, float)` by dividing each value by 255
+- [x] Task 2.2: Verify `Colors` enum values are correct (spot-check primaries, grays, edge cases)
 
 ### Verification
 
-- [ ] All existing `Colors`-related tests pass
-- [ ] Pyright passes on `color.py`
+- [x]All existing `Colors`-related tests pass
+- [x]Pyright passes on `color.py`
```
[the remark this answers is not in the document]
Comment (Djelibeybi) [reply conductor/tracks/hsbk-rgb-float_20260318/plan.md]: Fixed in `3105b6c` — added missing space after `[x]` on lines 39-40.

File: conductor/tracks/hsbk-rgb-float_20260318/plan.md [hunk 3]
```diff
@@ -15,68 +15,68 @@ Update `from_rgb()` and `to_rgb()` signatures and their direct tests.
...
+- [x] Task 3.1: Update docstring examples in `light.py`, `multizone.py`, `matrix.py`, `api.py`, `effects/base.py`, `effects/pulse.py`
+- [x] Task 3.2: Update `test_effects/test_pulse.py` to use float values
+- [x] Task 3.3: Run full test suite and fix any remaining breakage
 
 ### Verification
 
-- [ ] Full test suite passes
-- [ ] `ruff check` and `ruff format` pass
+- [x]Full test suite passes
+- [x]`ruff check` and `ruff format` pass
```
[the remark this answers is not in the document]
Comment (Djelibeybi) [reply conductor/tracks/hsbk-rgb-float_20260318/plan.md]: Fixed in `c34d967` — added missing space after `[x]` on lines 54-55.

File: conductor/tracks/hsbk-rgb-float_20260318/plan.md [hunk 4]
```diff
@@ -15,68 +15,68 @@ Update `from_rgb()` and `to_rgb()` signatures and their direct tests.
...
+- [x] Task 4.2: Replace `_rgb_to_hsb()` calls with `HSBK.from_rgb()`
+- [x] Task 4.3: Remove `_hsb_to_rgb()` and `_rgb_to_hsb()` functions and `_HUE_SEXTANTS` constant
+- [x] Task 4.4: Run fireworks tests to confirm no regressions
 
 ### Verification
 
-- [ ] Fireworks tests pass with identical behavior
-- [ ] No private RGB helpers remain in `fireworks.py`
+- [x]Fireworks tests pass with identical behavior
+- [x]No private RGB helpers remain in `fireworks.py`
```
[the remark this answers is not in the document]
Comment (Djelibeybi) [reply conductor/tracks/hsbk-rgb-float_20260318/plan.md]: Fixed in `7ac2ab4` — added missing space after `[x]` on lines 70-71.

File: conductor/tracks/hsbk-rgb-float_20260318/plan.md [hunk 5]
```diff
@@ -15,68 +15,68 @@ Update `from_rgb()` and `to_rgb()` signatures and their direct tests.
...
-- [ ] All acceptance criteria met
-- [ ] `uv run --frozen pytest` — full suite passes
-- [ ] `uv run pyright` — no type errors
-- [ ] `uv run ruff check . && uv run ruff format --check .` — clean
-- [ ] Ready for review
+- [x]All acceptance criteria met
+- [x]`uv run --frozen pytest` — full suite passes
+- [x]`uv run pyright` — no type errors
+- [x]`uv run ruff check . && uv run ruff format --check .` — clean
+- [x]Ready for review
```
[the remark this answers is not in the document]
Comment (Djelibeybi) [reply conductor/tracks/hsbk-rgb-float_20260318/plan.md]: Fixed in `b30d649` — added missing space after `[x]` on lines 75-79.

File: src/lifx/effects/fireworks.py [hunk 6]
```diff
@@ -500,14 +431,8 @@ def generate_frame(self, ctx: FrameContext) -> list[HSBK]:
...
-                h_deg, s_01, b_01 = _rgb_to_hsb(r_val, g_val, b_val)
                 colors.append(
-                    HSBK(
-                        hue=round(h_deg) % 360,
-                        saturation=round(s_01, 2),
-                        brightness=round(min(1.0, b_01), 2),
-                        kelvin=self.kelvin,
-                    )
+                    HSBK.from_rgb(r_val, g_val, b_val).with_kelvin(self.kelvin)
                 )
```
[the remark this answers is not in the document]
Comment (Djelibeybi) [reply src/lifx/effects/fireworks.py:436]: Good catch. Fixed in `978e7a9` — updated all four `with_*` helpers (`with_hue`, `with_saturation`, `with_brightness`, `with_kelvin`) to use the internal unrounded `_hue`/`_saturation`/`_brightness` fields instead of the rounded public properties. This preserves full float precision through the fireworks compositing pipeline and any other chained operations.

**Comment by Djelibeybi** [comment]
**Addressed suppressed low-confidence comment** (Copilot, `src/lifx/color.py:455`):

> `with_hue()` constructs the new HSBK using `self.saturation`/`self.brightness` properties, which now return rounded display values. This unintentionally quantizes the internal float state...

Fixed in `978e7a9` — all four `with_*` helpers (`with_hue`, `with_saturation`, `with_brightness`, `with_kelvin`) now use the internal unrounded `_hue`/`_saturation`/`_brightness` fields, preserving full float precision.
